Fix audio passthrough at display lost/reset events

thexai opened the pull request:
## Description
Backport of https://github.com/xbmc/xbmc/pull/19724


## What is the effect on users?
Fixes missing passthrough audio when Adjust Display Refresh Rate at Start/Stop is enabled (only some systems are affected).


## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

thexai committed:
Fix audio passthrough at display reset events

FernetMenta commented on xbmc/cores/VideoPlayer/VideoPlayer.cpp:
```diff
@@ -5152,3 +5152,8 @@ void CVideoPlayer::SetUpdateStreamDetails()
+
+void CVideoPlayer::TriggerCheckPassthrough()
+{
+  m_VideoPlayerAudio->SendMessage(new CDVDMsg(CDVDMsg::PLAYER_AUDIO_DEVICE_CHANGED), 1);
```
this is wrong and can even segfault. m_VideoPlayerAudio can only be accessed from VP's internal thread

thexai replied:
You have a lot of knowledge of Kodi but that does not mean that everything you say is necessarily 100% true all the time.

thexai commented:
Reverted to initial and more simple idea of display lost/reset events. This is sufficient to fix 90% of cases and the other 10% is also fixed in combination with "Delay after refresh rate change" setting to 2 - 3 seconds.

ksooo commented:
Not sure it is a good idea to ignore what @FernetMenta said. He seems to have concerns I I cannot see any meaningful points of the PR submitter to prove him wrong.

thexai commented:
> Not sure it is a good idea to ignore what @FernetMenta said. He seems to have concerns and I cannot see any meaningful points of the PR submitter to prove him wrong.

Not ignored, as said previous comment:

> Reverted to initial and more simple idea of display lost/reset events. This is sufficient to fix 90% of cases and the other 10% is also fixed in combination with "Delay after refresh rate change" setting to 2 - 3 seconds.

@FernetMenta comment refers to a previous code (reverted) and actual code is safe and already merged and approved in master branch. See: https://github.com/xbmc/xbmc/pull/19724

ksooo commented:
> comment refers to a previous code (reverted), current code is safe and already merged and approved in master branch.

ah. thanks for the clarification.

ksooo commented:
As this backport is a 100% code copy of the respective master PR I think this one can go in.
